woocommerce/woocommerce-android
# [Coupons] Load and show performance report

Closes: #5530


### Description
This PR adds logic for loading and displaying the performance report for a given coupon, I followed the same approach that iOS did in the PR: https://github.com/woocommerce/woocommerce-ios/pull/6171, meaning this:
1. The number of orders is displayed right away when we get the coupon from DB.
2. If the number of orders equals `0`, we will directly display `0` as amount as well, otherwise a loading indicator is shown.
3. If the loading of performance fails, we will show an error Snackbar.

### Testing instructions
Please note that the `usageCount` of the Coupon itself is still hardcoded [here](https://github.com/woocommerce/woocommerce-android/blob/f09104860583f30677d1641dda3221d477eb1278/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/coupons/details/CouponDetailsRepository.kt#L26), so to test different scenarios, please change its value.

1. Open the app and click on More tab.
2. Click on Coupons.
3. Click on one the coupon details.
4. Notice the value set in `usageCount` is displayed first in "Discounted orders", with a loading indicator in the amount (if the value != 0).
5. Notice the performance amount is displayed after loading, and the "Discounted orders" value is updated to the correct value.

### Images/gif
https://user-images.githubusercontent.com/1657201/161973047-7e8bbce3-d40f-4aff-8cfb-c6438fe30bbc.mp4
